store-api-server/src/payment/controller/payment.controller.ts

```diff
@@ -191,48 +192,71 @@ export class PaymentController {
     @Body('recreateNftOrder') recreateNftOrder: boolean = false,
     @Body('currency') currency: string = BASE_CURRENCY,
   ) {
-    const [firstProvider, secondProvider] = paymentProviders.map((provider) => {
+    const providers = paymentProviders.map((provider) => {
       return {
         provider,
         currency: this.paymentService.getCurrency(provider, currency),
       };
     });
 
-    this.userService
-      .handleCartAdd({ cookieSession, user, nftId })
-      .then(async () => {
-        const args = {
-          cookieSession,
-          user,
-          request,
-          recreateNftOrder,
-        };
-        const firstProviderIntent =
-          await this.paymentService.handleCreatePaymentIntent({
-            ...args,
-            paymentProvider: firstProvider.provider,
-            currency: firstProvider.currency,
-          });
-        const secondProviderIntent =
-          await this.paymentService.handleCreatePaymentIntent({
-            ...args,
-            paymentProvider: secondProvider.provider,
-            currency: secondProvider.currency,
-          });
-        return {
-          [firstProvider.provider]: firstProviderIntent,
-          [secondProvider.provider]: secondProviderIntent,
-        };
-      })
-      .catch((err) => {
-        Logger.error(
-          `Unable to create payment intents for payment providers, err: ${err}`,
-        );
+    try {
+      await this.userService.handleCartAdd({ cookieSession, user, nftId });
+
+      const args = {
+        cookieSession,
+        user,
+        request,
+        recreateNftOrder,
+      };
+
+      const firstProvider = providers.shift();
+      if (!firstProvider) {
         throw new HttpException(
-          'failed to create payment intents for payment providers',
+          'No providers provided',
           HttpStatus.INTERNAL_SERVER_ERROR,
         );
+      }
+      const firstIntent = await this.paymentService.handleCreatePaymentIntent({
+        ...args,
+        paymentProvider: firstProvider.provider,
+        currency: firstProvider.currency,
       });
+
+      const remainingIntents = await Promise.all(
+        providers.map(async (currentProvider) => {
+          const providerIntent =
+            await this.paymentService.handleCreatePaymentIntent({
+              ...args,
+              paymentProvider: currentProvider.provider,
+              currency: currentProvider.currency,
+            });
+
+          return {
+            [currentProvider.provider]: providerIntent,
+          };
+        }),
+      );
+
+      const allProviderIntents = [
+        ...[{ [firstProvider.provider]: firstIntent }],
```

Comment:
cant this line just be `{ [firstProvider.provider]: firstIntent },`?

Reply:
Array.reduce() only works on arrays. So it needs to be an array.

Reply:
You are correct, changed it.